### PR TITLE
Clarify documentation and properties of `splitCoin`. 

### PR DESCRIPTION
**Pull request by jonathanknowles**
## Related Issues

#18
#21
#22 

## Summary

This PR:

- [x] Clarifies the documentation of `splitCoin`, adding examples.
- [x] Tests that `splitCoin` preserves the total sum.
- [x] Tests that `splitCoin` only produces valid coins.

### CODE REVIEW

File: src/Cardano/Fee.hs [hunk 1]
```diff
@@ -445,18 +448,56 @@ remainingFee opts s = do
...
+--
+-- >>> splitCoin (Coin 10) (Coin <$> [45000000000000000])
+-- [Coin 45000000000000000, Coin 10]
+--
+-- >>> splitCoin (Coin 10) (Coin <$> [45000000000000000 - 1])
+-- [Coin 44999999999999999, Coin 10]
+--
+-- == Properties
+--
+-- The total value is preserved:
```
Comment (paweljakubas) [inline src/Cardano/Fee.hs:495]: :100: fr those examples

File: test/Cardano/FeeSpec.hs [hunk 2]
```diff
@@ -712,6 +721,79 @@ propRebalanceChangeOutputsPreservesSum
...
+            [ pure minBound
+            , pure maxBound
+            , Coin <$> choose
+                ( succ . getCoin $ minBound
+                , pred . getCoin $ maxBound
+                )
+            ]
+
+propSplitCoinDataCoverage :: SplitCoinData -> Property
+propSplitCoinDataCoverage (SplitCoinData coinToSplit coinsToIncrease) =
```
Comment (paweljakubas) [inline test/Cardano/FeeSpec.hs:755]: what about a situation when we have `cointToSplit = N` and the number of coins to be increased more than `N` ?

Comment (paweljakubas) [reply test/Cardano/FeeSpec.hs:755]: probably this coverage is missing ^^^

File: src/Cardano/Fee.hs [hunk 3]
```diff
@@ -445,18 +448,56 @@ remainingFee opts s = do
...
+--   integer, producing a new list of coin values where the overall total is
+--   preserved.
+--
+-- == Examples
+--
+-- >>> splitCoin (Coin 4) (Coin <$> [1, 1, 1, 1])
+-- [Coin 2, Coin 2, Coin 2, Coin 2]
+--
+-- >>> splitCoin (Coin 4) (Coin <$> [1, 2, 3, 4])
+-- [Coin 2, Coin 3, Coin 4, Coin 5]
```
Comment (jonathanknowles) [inline src/Cardano/Fee.hs:463]: Perhaps these examples could use a higher value for `coinToSplit`, to better illustrate that each coin is increased by an identical amount.

File: src/Cardano/Fee.hs [hunk 4]
```diff
@@ -445,18 +448,56 @@ remainingFee opts s = do
...
+--
+-- == Special Cases
+--
+-- While processing the given list, if increasing the value of any given coin
+-- 'c' would cause its value to exceed 'maxBound', this function will leave
+-- coin 'c' unchanged in the resulting list, distributing the excess value to
+-- coins that occur later in the list:
+--
+-- >>> splitCoin (Coin 10) (Coin <$> [45000000000000000, 1])
+-- [Coin 45000000000000000, Coin 11]
```
Comment (jonathanknowles) [inline src/Cardano/Fee.hs:473]: Perhaps there is a way of avoiding these long numerical literals within examples.

File: test/Cardano/FeeSpec.hs [hunk 5]
```diff
@@ -712,6 +721,79 @@ propRebalanceChangeOutputsPreservesSum
...
+      where
+        genCoin :: Gen Coin
+        genCoin = oneof
+            [ pure minBound
+            , pure maxBound
+            , Coin <$> choose
+                ( succ . getCoin $ minBound
+                , pred . getCoin $ maxBound
+                )
+            ]
```
Comment (jonathanknowles) [inline test/Cardano/FeeSpec.hs:752]: Perhaps add an implementation of `shrink`?

File: src/Cardano/Fee.hs (hunk 4 above)
Comment (jonathanknowles) [reply src/Cardano/Fee.hs:445]: Fixed in 6f5bd77559cb4d27d81ba71f2bee53dc761ada49.

File: src/Cardano/Fee.hs (hunk 3 above)
Comment (jonathanknowles) [reply src/Cardano/Fee.hs]: Fixed in 6f5bd77559cb4d27d81ba71f2bee53dc761ada49.

File: test/Cardano/FeeSpec.hs (hunk 5 above)
Comment (jonathanknowles) [reply test/Cardano/FeeSpec.hs:756]: Fixed in 7f61a54dd452a082f0a20e2f5120e4b9bd63b451.

File: test/Cardano/FeeSpec.hs (hunk 2 above)
Comment (jonathanknowles) [reply test/Cardano/FeeSpec.hs:762]: Fixed in the following commits:
3c3493b16b1d814e892e9a491b735fad3bddf5f5
90b7dbf9614116a00e57c9136d4b773c51235a05

**Pull request merged**